src/auth/integration-tests/tests/driver.rs

```diff
@@ -28,12 +28,16 @@ mod driver {
     #[cfg(all(test, feature = "run-byoid-integration-tests"))]
     #[tokio::test(flavor = "multi_thread", worker_threads = 1)]
     async fn run_workload_identity_provider_url_sourced() -> anyhow::Result<()> {
-        auth_integration_tests::workload_identity_provider_url_sourced().await
```

Comment:
use the `test_case` crate to pass the `with_impersonation` param. Would be something like:

```
#[test_case(false; "default")]
#[test_case(true; "with impersonation")]
async fn run_workload_identity_provider_url_sourced(with_impersonation: bool) -> anyhow::Result<()> {
  auth_integration_tests::workload_identity_provider_url_sourced(with_impersonation).await?;
}
```